src/Sarif/Writers/MimeType.cs

```diff
@@ -24,16 +25,18 @@ public static string DetermineFromFileExtension(string path)
                 return MimeType.Directory;
             }
 
-            foreach (ImmutableArray<string> tableEntry in s_extensionTable)
+            foreach (ImmutableArray<ImmutableArray<string>> extensionsTable in GetExtensionsTables())
```

Comment:
I know this is existing code, but... why are we crawling an array? Shouldn't we build a hash table?

Reply:
(We already have one for the textual MIME types.)

Reply:
is there a note here?

---
In reply to: [200730359](https://github.com/Microsoft/sarif-sdk/pull/936#discussion_r200730359) [](ancestors = 200730359)